src/SalePercentage.php

```diff
@@ -245,12 +245,21 @@ public static function getProductSalePercentage($product) {
    *
    * @param int $salePercentage
    *   The product sale percentage.
+   * @param \WC_Product $product
+   *   The product object.
    *
    * @return bool
    *   TRUE if the sale percentage flash bubble should be displayed.
    */
-  public static function checkDisplaySalePercentage($salePercentage) {
-    return (!is_single() && $salePercentage >= static::getMinimumSalePercentage()) || is_single();
+  public static function checkDisplaySalePercentage($salePercentage, \WC_Product $product) {
+    $product_category_ids = wp_get_post_terms($product->get_id(), 'product_cat', ['fields' => 'ids']);
+    $eligible_category_ids = get_option('_sale_percentage_eligible_product_categories') ?? [];
+    $eligible_categories = array_intersect($eligible_category_ids, $product_category_ids);
+    $is_category_eligible = !empty($eligible_categories);
```

Comment:
Maybe we don't even need a new variable here, and check directly as in next suggestion.
```suggestion
```

Reply:
@bogdanarizancu 
Resolved in the last commit